Dockerfile

```diff
@@ -1,7 +1,26 @@
+# Build stage
+##################################################
+FROM docker.io/golang:1.23-alpine3.20 AS build
```

Comment:
If you don't want to have to manually bump this from time to time, [there are other tags available](https://hub.docker.com/_/golang) in varying degrees of brevity.

It's a tradeoff between convenience and reproducibility. Your call.

Reply:
what is the image size difference between this and the current approach?

Reply:
> what is the image size difference between this and the current approach?

Basically identical, which is to be expected.

```
❯ podman images -n --format "{{.Repository}}:{{.Tag}} {{.Size}}" | rg tcping
localhost/tcping-new:latest 9.61 MB
ghcr.io/pouriyajamshidi/tcping:latest 9.62 MB
```